server/service/service.go

```diff
@@ -0,0 +1,95 @@
+package service
+
+import (
+	"context"
+
+	"github.com/openconfig/bootz/proto/bootz"
+	"github.com/openconfig/gnmi/errlist"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type EntityLookup struct {
+	Manufacturer string
+	SerialNumber string
+}
+
+type ChassisEntity struct {
+	BootMode string
+}
+
+type EntityManager interface {
+	ResolveChassis(*EntityLookup) (*ChassisEntity, error)
+	GetBootstrapData(*bootz.ControlCard) (*bootz.BootstrapDataResponse, error)
+	SetStatus(*bootz.ReportStatusRequest) error
+	Sign(*bootz.GetBootstrapDataResponse) error
+}
+
+type Service struct {
+	bootz.UnimplementedBootstrapServer
+	em EntityManager
+}
+
+func (s *Service) GetBootstrapRequest(ctx context.Context, req *bootz.GetBootstrapDataRequest) (*bootz.GetBootstrapDataResponse, error) {
+	if len(req.ChassisDescriptor.ControlCards) == 0 {
+		return nil, status.Errorf(codes.InvalidArgument, "request must include at least one control card")
+	}
+	lookup := &EntityLookup{
```

Comment:
i don't follow?

Reply:
It's not recommended to not use the field names here -- since if you add a new field then they aren't robust to it (it'll complain that it's not defined). There was a style guide update a while back.